tests/test_api.py

```diff
@@ -556,6 +556,17 @@ def run():
         t.start()
         t.join()
 
+def test_pool_close():
+    pool = ConnectionPool(size=3, **connection_kwargs)
+
+    with pool.connection():
+        with assert_raises(TypeError):
+            pool.close_connections(timeout='foo')
+
+        with assert_raises(ValueError):
+            pool.close_connections(timeout=-1)
+
+        pool.close_connections()
```

Comment:
this actually does not test that the logic works. :)  not sure about a sane test though.